### ISSUE TITLE
Add support for workspace specific java.home that use vscode variables

### ISSUE THREAD

**Issue by bajajh** [redhat-developer/vscode-java#806]
In the workspace specific `settings.json`,

Does not work:
`"java.home": "${workspaceRoot}/java/jdk64"`

Works:
`"java.home": "D:/dev/proj-1/workgroup-support/java/jdk64"` works.

##### Environment
- Operating System: Windows
- JDK version: 8
- Visual Studio Code version: 1.31.1
- Java extension version: 0.38.0

##### Steps To Reproduce
1. Open a workspace
1. Create a folder named `.vscode`
1. Create a file in the folder named `settings.json`
1. Try the setting above that does not work by adding it to `settings.json`

**Comment by fbricon** [comment]
@yaohaizh do you know if there's an API to resolve vscode properties?

**Comment by yaohaizh** [comment]
There is no automatically way to resolve this value from VSCode settings. We need provide the resolve logic by our extension.

**Comment by testforstephen** [comment]
yes, no official API for this yet. In java debug, we copied the vscode code https://github.com/Microsoft/vscode-java-debug/blob/master/src/variableResolver.ts

**Comment by hanct** [comment]
Hi,
Just curious, does this URL helps https://github.com/microsoft/vscode-java-debug/issues/750 ?
Quoted from this URL 

"In VS Code 1.42 we will introduce a new hook resolveDebugConfigurationWithSubstitutedVariables in the extension API that receives the debug configuration with all variables resolved."

**Comment by hanct** [comment]
I think someone has just released an extension to support variable in settings.  Hope vscode-java exntesions developer can consider using it to support relative path for "java.home", etc
[https://www.npmjs.com/package/vscode-variables](https://www.npmjs.com/package/vscode-variables)

**Comment by srl295** [comment]
Any update on this? Also, variables work in `java.test.config`:

```json
    "java.test.config": [{
        "name": "cldrtest",
        "workingDirectory": "${workspaceFolder}",
        "vmargs": [
            "-DCLDR_DIR=${workspaceFolder}",
            "-DCLDR_ENVIRONMENT=UNITTEST"
        ]
    }],
```

… but not in `vmArgs`:

```json
{
    "java.debug.settings.vmArgs": "-DCLDR_DIR=${workspaceFolder}",
}
```
